src/main/java/com/rackspace/salus/event/manage/services/TasksService.java

```
@@ -202,4 +150,114 @@ public void deleteAllTasksForTenant(String tenant) {
     eventEngineTaskRepository.findByTenantId(tenant, Pageable.unpaged())
         .forEach(task -> deleteTask(tenant, task.getId()));
   }
+
+  @Transactional
+  public EventEngineTask updateTask(@Valid EventEngineTask update) {
```

This also relates to my comment below regarding the use of `convertFromInput`, as I don't think `@Valid EventEngineTask` is ideal here with how you've got things set up.

e.g. `CreateTask` could contain a `null` measurement field, which means `update` would now have a `null` measurement which is invalid.  I don't think `@Valid` will actually fail since the null setting is in the column annotation instead of `@NotNull` but the intent is the same.

(@itzg this always brings up a point of maybe we should be using the validation annotations in addition to column nullables on our entities)

---

@jjbuchan yeah, now that @rohitsngh27 [figured out how to intercept and provide better error reporting on JPA constraint violations](https://github.com/racker/salus-telemetry-monitor-management/pull/197/files#diff-03f139be4207edb3b9d87c275c560960R68-R77) then I agree that we could now add more of those declarations to the entities.